components/Msig/CreateConfirm.tsx

```diff
@@ -0,0 +1,154 @@
+
+export const CreateConfirm = () => {
+  const [msigError, setMsigError] = useState(false)
+  const [confirming, setConfirming] = useState(false)
+  const { setMsigActor, Address } = useMsig()
+  const router = useRouter()
+  const { cid } = router.query
+  const { NEXT_PUBLIC_EXPLORER_URL: explorerUrl } = process.env
+  const { NEXT_PUBLIC_LOTUS_NODE_JSONRPC: apiAddress } = process.env
+
+  const confirm = useCallback(async () => {
+    try {
+      const confirmed = await confirmMessage(cid as string, { apiAddress })
+      if (confirmed) {
+        const lCli = new LotusRPCEngine({ apiAddress })
+        const receipt = await lCli.request<{
+          ExitCode: number
+          GasUsed: number
+          Return: string
+        }>('StateGetReceipt', { '/': cid }, null)
+
+        // Verify exit code
+        if (receipt.ExitCode !== 0) {
+          setMsigError(true)
+          return
+        }
+
+        // Verify address
+        const { robust } = getAddrFromReceipt(receipt.Return)
+        if (!robust) {
+          setMsigError(true)
+          return
+        }
+
+        // Set safe address
+        setMsigActor(robust)
+      }
+    } catch (e) {
+      setMsigError(true)
+    }
+  }, [cid, apiAddress, setMsigActor])
+
+  useEffect(() => {
+    if (!Address && !confirming && cid) {
+      setConfirming(true)
+      confirm()
+    }
+  }, [cid, Address, confirming, setConfirming, confirm])
+
+  // CID was not provided in query parameters
+  if (!cid) {
+    return (
+      <Dialog>
+        <ShadowBox>
+          <h2>Safe creation status unknown</h2>
+          <hr />
+          <p>Missing CID for monitoring safe creation</p>
+        </ShadowBox>
+      </Dialog>
+    )
+  }
+
+  // Something went wrong while creating the safe
+  if (msigError) {
+    return (
+      <Dialog>
+        <ShadowBox>
+          <h2>Safe creation failed</h2>
+          <hr />
+          <p>Something went wrong</p>
+        </ShadowBox>
+        <ErrorBox>
+          <p>
+            An error occurred while creating your safe. Click on the CID below
+            to view the transaction in the Glif Explorer:
+          </p>
+          <p>
+            <SmartLink href={`${explorerUrl}/message/?cid=${cid}`}>
+              {cid}
+            </SmartLink>
+          </p>
+        </ErrorBox>
+      </Dialog>
+    )
+  }
+
+  // The safe has been successfully created
+  if (Address) {
+    return (
+      <Dialog>
+        <ShadowBox>
+          <h2>Safe creation completed</h2>
+          <hr />
+          <p>Your new Safe address is:</p>
+          <p>
+            <SmartLink href={`${explorerUrl}/actor/?address=${Address}`}>
+              {Address}
+            </SmartLink>
+          </p>
+        </ShadowBox>
+        <ButtonRowCenter>
+          <ButtonV2
+            large
+            green
+            onClick={() => navigate(router, { pageUrl: PAGE.MSIG_HOME })}
+          >
+            Go to Safe dashboard
+          </ButtonV2>
```

Comment:
The current solution now shows a button that says: "Connect your wallet to open the Safe"
Also there's a warning box saying: "Please make sure to store your Safe address before continuing to prevent losing access to your Safe."

It still seems very easy to lose access, do you think it would be possible to create a page where we display all the safe addresses that the user created? We might achieve this by looking at the message history?